Tests/SPMBuildCoreTests/PluginInvocationTests.swift

```diff
@@ -294,20 +314,140 @@ class PluginInvocationTests: XCTestCase {
             // Try to compile the fixed plugin. This time it should succeed but we expect a warning.
             let result = try tsc_await { pluginScriptRunner.compilePluginScript(
                 sources: buildToolPlugin.sources,
-                toolsVersion: .currentToolsVersion,
+                toolsVersion: buildToolPlugin.apiVersion,
                 observabilityScope: observability.topScope,
                 callbackQueue: DispatchQueue(label: "plugin-compilation"),
                 completion: $0) }
             
             // Now we expect compilation to succeed but with a warning.
-            XCTAssert(result.compilerResult.exitStatus == .terminated(code: 0), "\(result.compilerResult.exitStatus)")
+            XCTAssert(result.wasCached == false)
+            XCTAssert(result.compilerResult?.exitStatus == .terminated(code: 0), "\(String(describing: result.compilerResult?.exitStatus))")
             XCTAssert(result.compiledExecutable.components.contains("plugin-cache"), "\(result.compiledExecutable.pathString)")
             XCTAssert(result.diagnosticsFile.suffix == ".dia", "\(result.diagnosticsFile.pathString)")
             let contents = try localFileSystem.readFileContents(result.diagnosticsFile)
             let diags = try SerializedDiagnostics(bytes: contents)
             XCTAssertEqual(diags.diagnostics.count, 1)
             let warningDiag = try XCTUnwrap(diags.diagnostics.first)
             XCTAssertTrue(warningDiag.text.hasPrefix("variable \'unused\' was never used"), "\(warningDiag)")
+
+            // Recompile the command plugin again without changing its source code.
+            let recompilationResult = try tsc_await { pluginScriptRunner.compilePluginScript(
```

Comment:
nit: create a test only private extension for PluginScriptRunner that is an sync, throwing version of compilePluginScript (which does the `try tsc_await` dance)

Reply:
Sounds good.  Fixed.

Reply:
I also cleaned up the rest of the test to be more readable.  Could use some refactoring of the test, but will do in a later PR.